great_expectations/core/expectation_configuration.py

```diff
@@ -1259,24 +1283,24 @@ def isEquivalentTo(
         if match_type == "domain":
             return all(
                 (
-                    self.expectation_type == other.expectation_type,
-                    self.get_domain_kwargs() == other.get_domain_kwargs(),
+                    self.expectation_type == other.expectation_type,  # type: ignore[union-attr] # could be dict
+                    self.get_domain_kwargs() == other.get_domain_kwargs(),  # type: ignore[union-attr] # could be dict
```

Comment:
Of other could be a dict why not use `other.expectation_type or other.get("expectation_type")`?

Reply:
I'd prefer to save logic changes for another PR.